src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -215,31 +192,97 @@ public void init(Properties properties) {
     }
 
     /**
-     * Returns the secret corresponding to the given alias name
+     * Check whether to use the provider listed under secretRepositories
+     * or secretProviders property for resolving secrets.
      *
-     * @param alias The logical or alias name
-     * @return If there is a secret , otherwise , alias itself
+     * @param secretAnnotation String contains the alias, the provider type and the repository type.
+     * @return plain text value for the required secret.
+     */
+    public String resolveSecret(String secretAnnotation) {
+
+        String[] secretAnnotationStrings = secretAnnotation.split(DELIMITER);
+        if (secretAnnotationStrings.length == 1) {
+            if (isLegacyProvidersExists) {
+                return getSecret(secretAnnotation);
+            }
+        }
+        return resolveSecret(secretAnnotationStrings);
+    }
+
+    /**
+     * Resolve the secret annotation for the secrets coming from
+     * repositories belongs to providers listed under secretProviders property.
+     *
+     * @param annotation vaule retrieve by the resolveSecret as the value to be resolved.
+     * @return If there is a secret , otherwise , alias itself.
+     */
+    public String resolveSecret(String[] annotation) {
+
+        int length = annotation.length;
+        try {
+            switch (length) {
+                case 1:
+                    return getSecret((String) providers.keySet().toArray()[0],
```

Comment:
better to add a comment explaining bit about this logic

Reply:
Added a descriptive comment